File: Ultrasonic_waves/sr501.py
import RPi.GPIO as GPIO
import time

GPIOIN = 17
GPIOOUT = 27

GPIO.setmode(GPIO.BCM)
print("detection start")

GPIO.setup(GPIOIN, GPIO.IN) #센서
GPIO.setup(GPIOOUT, GPIO.OUT) #출력 LED

try:
    while True: # 센서 출력 값을 읽음
        state = GPIO.input(GPIOIN)
        if(state == True):
            print("모션 감지")
        else:
            print("모션 감지 실패")
        GPIO.output(GPIOOUT, state) #센서 검출 값을 LED로
        time.sleep(1)
except KeyboardInterrupt:
    GPIO.cleanup()
print("END!!!")

File: Ultrasonic_waves/sr501.c
#include <wiringPi.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

#define GPIOIN 17
#define GPIOOUT 27

void read(){
    uint8_t State;
    pinMode(GPIOIN, INPUT);
    pinMode(GPIOOUT, OUTPUT);
    printf("Sensor start!!");

    while(1){
        delay(1000);
        State = digitalRead(GPIOIN);
        printf("state = %d", State)
        digitalWrite(GPIOOUT, State) //LED 켜거나 끔
    }
    
}

int main(void){
    if(wiringPiSetupGPIO() == -1) {
        exit(1);
    }
    read()
    return(0);
}